src/main/java/org/moltimate/moltimatebackend/service/GenerateMotifService.java

```diff
@@ -33,13 +33,56 @@
     @Autowired
     private ProteinService proteinService;
 
+
+    public Integer threadUpdateMotifs() {
+        log.info("Updating Motif database");
+        List<ActiveSite> activeSites = activeSiteService.getActiveSites();
+
+        log.info("Deleting and flushing Motif database");
+//        motifService.deleteAllAndFlush();
+
+        log.info("Saving " + activeSites.size() + " new motifs");
+        AtomicInteger motifsSaved = new AtomicInteger(0);
+        List<String> failedPdbIds = new ArrayList<>();
+        activeSites.parallelStream()
+                .forEach(activeSite -> {
+                    String pdbId = activeSite.getPdbId();
+                    try {
+                        List<Residue> residues = activeSite.getResidues();
+                        Structure structure = proteinService.queryPdb(pdbId);
+                        for (Residue res : residues) {
+                            Group group = StructureUtils.getResidue(
+                                    structure, res.getResidueName(), res.getResidueId());
+                            res.setResidueChainName(group.getResidueNumber()
+                                    .getChainName());
+                            res.setResidueAltLoc(Residue.getAltLocFromGroup(group));
+                        }
+                        Motif motif = Motif.builder()
+                                .pdbId(pdbId)
+                                .activeSiteResidues(residues)
+                                .ecNumber(StructureUtils.ecNumber(structure))
+                                .selectionQueries(generateSelectionQueries(structure, residues))
+                                .build();
+                        motifService.saveMotif(motif);
+                        motifsSaved.incrementAndGet();
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                        failedPdbIds.add(pdbId);
+                    }
+                });
+
+        log.info("Failed to save " + failedPdbIds.size() + " motifs to the database: " + failedPdbIds.toString());
+        log.info("Finished saving " + motifsSaved.get() + " motifs to the database");
+        return motifsSaved.get();
+    }
+
     // TODO: Pessimistic lock the motifs table until update is finished
     public void updateMotifs() {
```

Comment:
can we not have both of these, now that we know it works can we keep `threadUpdateMotifs` implementation with the returned number and change its name to `updateMotifs`